### PR TITLE
migrate to android studio 3

### PR DESCRIPTION
**Pull request by IlyaEremin**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -109,6 +107,7 @@ android {
     configurations.all {
         resolutionStrategy.force "com.android.support:support-annotations:$versions.support"
     }
+    buildToolsVersion '26.0.2'
```
Comment (ValeriyMakarshin) [inline app/build.gradle]: Duplication with line 19

File: build.gradle [hunk 2]
```diff
@@ -1,11 +1,11 @@
 buildscript {
     repositories {
         jcenter()
+        google()
     }
 
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.3.2'
-        classpath 'me.tatarka:gradle-retrolambda:3.2.5' // use Java 8 lambdas on android
```
Comment (ValeriyMakarshin) [inline build.gradle]: Maybe need to fix the mention of retrolambda in README.md, jacoco.gradle, lint.xml and proguard-rules.pro?

File: app/build.gradle (hunk 1 above)
Comment (ValeriyMakarshin) [inline app/build.gradle]: Duplication with line 19 (fix [deps.gradle](https://github.com/fs/android-base/blob/master/deps.gradle))

Comment (ValeriyMakarshin) [reply app/build.gradle]: Need update the version for ci

File: build.gradle (hunk 2 above)
Comment (IlyaEremin) [reply build.gradle]: 👍

**Comment by IlyaEremin** [comment]
Need to fix UI tests on CircleCI 😞

File: app/build.gradle [hunk 3]
```diff
@@ -118,18 +116,17 @@ repositories {
...
-    testCompile unitTestLibs
-    androidTestCompile androidTestsLibs
+    implementation supportLibs
+    implementation rxJavaLibs
+    //    implementation retrofitLibs
+    //    implementation okHttpLibs
+
+    compileOnly 'org.jetbrains:annotations:13.0'
+    implementation 'com.google.code.gson:gson:2.4'
+    implementation 'com.github.bumptech.glide:glide:3.7.0' // view injection
```
Comment (ValeriyMakarshin) [inline app/build.gradle]: Fix comment pls

File: app/build.gradle [hunk 4]
```diff
@@ -25,25 +24,24 @@ android {
 
         applicationId APPLICATION_ID
         versionCode 1
-        versionName '1.0-beta'
+        versionName '0.9.1'
         testInstrumentationRunner 'android.support.test.runner.AndroidJUnitRunner'
         com.android.ddmlib.DdmPreferences.setTimeOut(60000)
+        javaCompileOptions.annotationProcessorOptions.includeCompileClasspath = true
```
Comment (MaratIskhakov) [inline app/build.gradle]: Is this parameter really necessary there? 
> If you are experiencing issues migrating to the new dependency resolution strategy, you can restore behavior to that of Android plugin 2.3.0 by setting includeCompileClasspath true. However, restoring behavior to version 2.3.0 is not recommended, and the option to do so will be removed in a future update.

**Comment by IlyaEremin** [comment]
@ValeriyMakarshin Я отключу спун чтобы CircleCI не давал ошибку. Хочешь прикрутить composer вместо ложки?

File: app/build.gradle (hunk 4 above)
Comment (IlyaEremin) [reply app/build.gradle]: Yes. let's update butterknife in different PR

File: deps.gradle [hunk 5]
```diff
@@ -1,11 +1,11 @@
 ext {
     versions = [
             okHttp             : '3.3.1',
-            support            : '25.3.1',
+            support            : '27.0.1',
```
Comment (ValeriyMakarshin) [inline deps.gradle]: pls update [test-deps.gradle](https://github.com/fs/android-base/blob/6d505de8655af27b5dd6dcd951b3c2347f0c569e/test-deps.gradle#L3)

**Comment by IlyaEremin** [comment]
fuck fuck fuck hate circle ci

File: deps.gradle (hunk 5 above)
Comment (IlyaEremin) [reply deps.gradle]: 👍

**Comment by IlyaEremin** [comment]
@ValeriyMakarshin заапрувь и в путь

**Comment by IlyaEremin** [comment]
finally

**Pull request merged**